lib/analysis/SemanticValueSearch/ValueSearch.h

```diff
@@ -0,0 +1,237 @@
+//
+// Created by Jon Stephens on 2/4/23.
+//
+
+#ifndef VANGUARD_VALUESEARCH_H
+#define VANGUARD_VALUESEARCH_H
+
+#include <Graphs/SVFG.h>
+#include <SVF-FE/SVFIRBuilder.h>
+#include <SVF-FE/LLVMUtil.h>
+#include "SABER/SrcSnkDDA.h"
+#include "SVF-FE/LLVMModule.h"
+#include <Graphs/SVFGNode.h>
+#include "llvm/IR/Value.h"
+#include "analysis/taint-analysis/TaintAnalysis.h"
+
+namespace vanguard {
+    class ValueSearchGuide {
+    public:
+        virtual std::vector<SVF::CxtDPItem> next(const SVF::ContextCond *ctx, const SVF::VFGNode *node, const SVF::SVFGEdge *edge) {
+            if(const auto *addrStmt = SVF::SVFUtil::dyn_cast<SVF::AddrSVFGNode>(node)) {
+                return process(ctx, addrStmt, edge);
+            }
+            else if(const auto *copyStmt = SVF::SVFUtil::dyn_cast<SVF::CopySVFGNode>(node)) {
+                return process(ctx, copyStmt, edge);
+            }
+            else if(const auto *gepStmt = SVF::SVFUtil::dyn_cast<SVF::GepSVFGNode>(node)) {
+                return process(ctx, gepStmt, edge);
+            }
+            else if(const auto *loadStmt = SVF::SVFUtil::dyn_cast<SVF::LoadSVFGNode>(node)) {
+                return process(ctx, loadStmt, edge);
+            }
+            else if(const auto *storeStmt = SVF::SVFUtil::dyn_cast<SVF::StoreSVFGNode>(node)) {
+                return process(ctx, storeStmt, edge);
+            }
+            else if(const auto *formalIn = SVF::SVFUtil::dyn_cast<SVF::FormalINSVFGNode>(node)) {
+                return process(ctx, formalIn, edge);
+            }
+            else if(const auto *formalOut = SVF::SVFUtil::dyn_cast<SVF::FormalOUTSVFGNode>(node)) {
+                return process(ctx, formalOut, edge);
+            }
+            else if(const auto *actualIn = SVF::SVFUtil::dyn_cast<SVF::ActualINSVFGNode>(node)) {
+                return process(ctx, actualIn, edge);
+            }
+            else if(const auto *actualOut = SVF::SVFUtil::dyn_cast<SVF::ActualOUTSVFGNode>(node)) {
+                return process(ctx, actualOut, edge);
+            }
+            else if(const auto *mssaphi = SVF::SVFUtil::dyn_cast<SVF::MSSAPHISVFGNode>(node)) {
+                return process(ctx, mssaphi, edge);
+            }
+            else if(const auto *formalParam = SVF::SVFUtil::dyn_cast<SVF::FormalParmSVFGNode>(node)) {
+                return process(ctx, formalParam, edge);
+            }
+            else if(const auto *actualParam = SVF::SVFUtil::dyn_cast<SVF::ActualParmSVFGNode>(node)) {
+                return process(ctx, actualParam, edge);
+            }
+            else if(const auto *formalRet = SVF::SVFUtil::dyn_cast<SVF::FormalRetSVFGNode>(node)) {
+                return process(ctx, formalRet, edge);
+            }
+            else if(const auto *actualRet = SVF::SVFUtil::dyn_cast<SVF::ActualRetSVFGNode>(node)) {
+                return process(ctx, actualRet, edge);
+            }
+            else {
+                throw runtime_error("error");
```

Comment:
This error message is not very useful, maybe something more descriptive?
```suggestion
                throw runtime_error("unhandled SVF node type");
```